### PR TITLE
FLEX-4503 ~ Fixes erroneous update of GPS coordinates.

### PR DESCRIPTION
**Pull request by kevinsmeets**
When GPS coordinates are empty (null, null), an update of the device data persisted the values 0, 0.

### CODE REVIEW

**Comment by CLAAdmin** [comment]
[CLA assistant check](http://cla.opensmartgridplatform.org:80/OSGP/open-smart-grid-platform?pullRequest=16) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

File: integration-tests/cucumber-tests-platform-common/src/test/java/org/opensmartgridplatform/cucumber/platform/common/support/ws/core/CoreDeviceManagementClient.java [hunk 1]
```diff
@@ -149,4 +151,11 @@ public UpdateDeviceCdmaSettingsResponse getUpdateDeviceCdmaSettingsResponse(
         return (UpdateDeviceCdmaSettingsResponse) wst.marshalSendAndReceive(asyncRequest);
     }
 
+    public UpdateDeviceResponse updateDevice(final UpdateDeviceRequest request)
+            throws WebServiceSecurityException, GeneralSecurityException, IOException {
```
Comment (rlemmers) [inline integration-tests/cucumber-tests-platform-common/src/test/java/org/opensmartgridplatform/cucumber/platform/common/support/ws/core/CoreDeviceManagementClient.java]: Can we change this somehow, so we throw only one exception? Sonar checks "Public methods should throw at most one checked exception".

File: integration-tests/cucumber-tests-platform-common/src/test/resources/features/common/osgp-adapter-ws-core/DeviceInstallation/UpdateDevice.feature [hunk 2]
```diff
@@ -23,11 +23,37 @@ Feature: CoreDeviceInstallation Device Updating
...
-  #		| Alias                | AfterTest         |
+      | Alias                | AfterTest         |
+
+  Scenario: Updating device data does not change GPS coordinates ( FLEX-4503 )
+    Given a device
+      | DeviceIdentification | TEST1024000000001 |
+      | Alias                | Alias             |
+      # Default values for the GPS coordinates are null.
+    When receiving an update device request
+      | DeviceIdentification      | TEST1024000000001       |
```
Comment (rlemmers) [inline integration-tests/cucumber-tests-platform-common/src/test/resources/features/common/osgp-adapter-ws-core/DeviceInstallation/UpdateDevice.feature]: Can we remove some of the fields? If they're not essential or interesting for this feature, we can use the defaults.

File: integration-tests/cucumber-tests-platform-common/src/test/resources/features/common/osgp-adapter-ws-core/DeviceManagement/UpdateDevice.feature [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      | DeviceIdentification | TEST1024000000001 |
+      | Alias                | BeforeTest        |
+    When receiving a device management update device request
+      | DeviceIdentification      | TEST1024000000001       |
+      | Alias                     | AfterTest               |
+      | NetworkAddress            | 127.0.0.1               |
+      | internalId                |                       1 |
+      | externalId                |                       2 |
+      | relayType                 | LIGHT                   |
+      | code                      |      100000000000000000 |
```
[the remark this answers is not in the document]
Comment (kevinsmeets) [reply integration-tests/cucumber-tests-platform-common/src/test/resources/features/common/osgp-adapter-ws-core/DeviceManagement/UpdateDevice.feature]: It appears to correlate with `org.opensmartgridplatform.adapter.ws.schema.core.devicemanagement.LightMeasurementDevice.code`.

**Pull request merged**